torch/csrc/jit/ir.h

```diff
@@ -220,6 +220,14 @@ struct Node : public Attributes<Node> {
   // Result:  %3 = f(%1, %2, %4)
   Node* addInput(Node * node) {
     JIT_ASSERT(graph_ == node->graph_);
+    // Select invariant
+    if (kind_ == kSelect) {
+      // You have no excuse for not having accurate type information on
+      // multi-return
+      JIT_ASSERT(node->hasType() && node->type()->kind() == TypeKind::MultiType);
+    } else {
+      JIT_ASSERT(!node->hasType() || node->type()->kind() != TypeKind::MultiType);
+    }
```

Comment:
Shouldn't this be only part of `lint()`?

Reply:
I ended up putting it both places; it's an easy invariant to check quickly, and it's better to fail quickly.